cmd/auth-provider-gcp/app/getcredentials_test.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+func TestValidateAuthFlow(t *testing.T) {
+	type FlagResult struct {
+		Name  string
+		Flow  string
+		Error error
+	}
+	tests := []FlagResult{
+		{Name: "validate gcr auth flow", Flow: gcrAuthFlow},
+		{Name: "validate docker-cfg auth flow option", Flow: dockerConfigAuthFlow},
+		{Name: "validate docker-cfg-url auth flow option", Flow: dockerConfigURLAuthFlow},
+		{Name: "bad auth flow option", Flow: "bad-flow", Error: &AuthFlowFlagError{flagValue: "bad-flow"}},
+		{Name: "empty auth flow option", Flow: "", Error: &AuthFlowFlagError{flagValue: ""}},
+		{Name: "case-sensitive auth flow", Flow: "Gcrauthflow", Error: &AuthFlowFlagError{flagValue: "Gcrauthflow"}},
```

Comment:
There are two related (but !=) tests related to this which should be covered. One is that the message is correct, for which getting the flagValue to match is important. There is a second test for Is() where we ensure that the flagValue is *NOT* used for the Is() check. For that test the bad flow value used to incur the error should not match the flagValue used to check the error in the Is() call.

Reply:
Gotcha, done